repairnator/kubernetes-support/README.md

```diff
@@ -0,0 +1,171 @@
+
+The starting folder for these deployment below is "repairnator/kubernetes-support".
+## Setup ActiveMQ
+
+To deploy ActiveMQ for effective queue managements of jobs to the pipeline and scanner. First apply the yaml file inside the queue-for-buildids folder
+```
+  cd repairnator/kubernetes-support/
+  kubectl create -f /queue-for-buildids/activemq.yaml
+```
+To access to the web interface
+```
+  kubectl get pods 
+```
+It should look like activemq-XXXXXXX-XXXXX. Then 
+```
+  kubectl exec -it activemq-XXXXXXX-XXXXX bash
+```
+To expose to localhost for webinterface access and also publishing(sending message to queue) outside the cluster. 
+```
+  kubectl port-forward activemq-XXXXXXX-XXXXX 8161:8161 61613:61613
+```
+Access webinterface with "http://localhost:8161/admin/" in your browser. The default username is "admin" and password is also "admin".
+
+To send message to queue use the publisher.py script in queue-for-buildids folder. Syntax
+```
+	cd repairnator/kubernetes-support/
+	python publisher.py -d queue-name message-1 message-2 message3 ... message-N
+```
+
+queue-name is according to the format /queue/name. For instance /queue/scanner will send message to the scanner queue.
+You can use this later to manually input build id to the pipeline queue or project to the scanner queue
+
+## Setup scanners and pipelines
+
+To deploy the scanner and pipeline, you first need to visit the yaml folder and provide some enviroment variables there and then apply the yaml file. As cmd lines, these steps are described as below
+
+```
+  cd repairnator-deployment-yamlfiles
+  kubectl create -f repairnator-worker.yaml
+  kubectl create -f repairnator-scanner.yaml
+```
+
+Note that the image specified in the yaml files does not need to be changed, since they are the main images for these components on repairnator official docker repository unless you want to build your own image. If that's the case , docker build, tag and push then apply replace your tagged image name with our in the yaml files. Now that's done we can now monitor our pods deployed
+
+```
+  kubectl get pods
+  kubectl logs -f repairnator-worker-XXXXXXXXX-XXXXX
+  kubectl logs -f repairnator-scanner-XXXXXXXXX-XXXXX
+```
+
+the first line "kubectl get pods" will give us the name of pod name of the worker and scanner for logging using the second and third line. To check that this is initilized correcly we can visit Activemq queue page 
+at "http://localhost:8161/admin/queues.jsp" then we should see that there is one consumer in both the scanner and the pipeline .
+
+## Example run
+In the folder "queue-for-buildids" use publisher.py to send messages to the queue. If you have a slug (like "surli/failingProject") you can send it to the scanner queue for scanning. The scanner will auto submit the interesting BuildIds to the pipeline queue for generating patches. So sending by
+```
+	python publisher.py /queue/scanner Tailp/travisplay
```

Comment:
Please update the format of this command, the latest version needs a `-d` option to specify the queue name.

Reply:
Fixed.